src/main/java/camp/nextstep/edu/calculator/StringCalculator.java

```diff
@@ -0,0 +1,20 @@
+package camp.nextstep.edu.calculator;
+
+import camp.nextstep.edu.calculator.StringSplitStrategy.StringSplitStrategyFactory;
+import java.util.List;
+
+public class StringCalculator {
+
+    public long calculateSum(String input) {
+        if(input == null || input.isEmpty()){
+            return 0L;
+        }
+        List<String> splitStrings =  StringSplitStrategyFactory.getStrategy(input).apply(input);
```

Comment:
디미터 원칙 위반입니다.
classA.methodA의 최대 지식한계는 아래와 같습니다. 
- classA의 필드 객체
- methodA가 생성한 객체
- methodA의 인자로 넘어온 객체 

다른 객체 내부의 데이터가 무엇이 있는지까지 알아서는 안되며,
역할을 담당하고 있는 객체에 메시지를 보내서 응답받도록 작성해보세요.

https://developerfarm.wordpress.com/2012/01/30/object_calisthenics_5/

---

가령, 이 경우엔 아래와 같이 작성해보는 것은 어떨까요

```suggestion
        List<String> splited =  StringUtils.split(input);
```